### ISSUE TITLE
General feedback #3

### ISSUE THREAD

**Issue by seblindfors** [seblindfors/ConsolePort#6]
Use this thread to post general feedback. What you should post here:
- Constructive general feedback
- Wanted changes
- Wanted additions
- Rants about terrible implementations

What you shouldn't post here:
- Bug reports
- Marketing concerns

**Comment by p0s1t** [comment]
Hi, great addon.
I'm going to try and help out. My first task will be to get the controller working in pet battles, which should be pretty easy. This is my interface at the moment. Sexymap for the map, blizz actionbars, moveanything, titanbar, wa2 and archud.
![wowscrnshot_081615_135220](https://cloud.githubusercontent.com/assets/13815732/9291903/1a0df1f2-441e-11e5-8be6-cf857e7dfd15.jpg)

**Comment by OxydBCN** [comment]
Hello, just like to point that nowadays im playing with Input Mapper, from the people of DS4Windows, wich offers almost zero lag peaks when using the ps4 controller wirelesly.

I did a port of your basic key bindings, but i cant make the "*" button properly, if i het it working i can uploaf the profile

**Comment by OxydBCN** [comment]
... Cant make the "*" button work properly, if i get it working... (Seems i cant edit in phone)

**Comment by seblindfors** [comment]
@OxydBCN  Input Mapper is probably just another branch of the originally open source software. I don't have the problems with DS4Windows you describe, but it's always nice with extra profiles. You don't HAVE to use NUMMULTIPLY or whatever it's called. It was frankly just the first button that came to mind that isn't normally used and isn't one of the function keys.

@p0s1t Pet battle support is coming with the next patch (a pretty big revamp of the addon btw) which I will release today or tomorrow.

**Comment by seblindfors** [comment]
Going to sort out a few bugs before releasing an overhaul to how the addon handles the interface. This update will add support for almost every standard Blizzard frame and will remove about half of the old code since it has become redundant using this new approach. Having learned a few tricks along the way of meticulously devouring the API, it was time I updated the addon to reflect what I've learned thus far.

This update will add support for:
- Almost all frames (might have forgotten to add a odd frames)
- Drop down menus
- Interface and system options
- Addon management
- Calendar
- Guild interface
- World map browsing
- Vendor and repairing
- PvP and PvE finder
- Pet battles
- Taint-free glyph, spec and talent picking
- Adventure guide and dungeon journal
- Collections (mounts, pets, toys, heirlooms)
- Garrison missions and garrison workorders
- Transmogrification
- Macro picking
- Professions
- Trainers
- Void storage
- Achievements
- Archaeology

A lot of things will be removed, and some things might be different to what you have gotten used to. Feedback on quirky stuff is highly appreciated. If there are any frames that lack support, please feel free to report their names or post screenshots and I'll fix the problem asap.

**Comment by OxydBCN** [comment]
Cant wait!

Question: With DS4Windows, your mouse moves smooth over the screen usign the controller wirelesly? Maybe the problem is my Bluetooth.

About the lighting, maybe putting the light a bit far away to reduce the contrast between light and shadow will help getting a more balanced image while keeping the dramatic badass look (light beneath character)

**Comment by seblindfors** [comment]
Yes, my mouse is working perfect with wireless. I'm using a Trust Bluetooth dongle.
I will try moving the light when I'm done with fixing bugs for the patch.

**Comment by seblindfors** [comment]
New patch is uploaded and relatively bug free. The new approach to handling the interface is a system that scans the interface for interactive buttons and then calculates the most appropriate action depending on which arrow you press on the controller. A fake cursor moves around and snaps to the interface buttons. This essentially allows the user to control any frame with ease, rather than trying to move the actual cursor around.

Pet battle example:
![image](https://cloud.githubusercontent.com/assets/7464938/9503627/723a789c-4c38-11e5-95e6-fdc042f6c53f.png)
Glyph frame example:
![image](https://cloud.githubusercontent.com/assets/7464938/9503639/8630be7e-4c38-11e5-87d1-9b275f7b8e83.png)

**Comment by OxydBCN** [comment]
Awesome approach!

Here's my first feedback:

Overall:
The user can realize pretty quick the behave of that new cursor.
My concerns are two:

- If the user will have trouble understanding that the new cursor is operated by the arrows, while the old one (and aestethically similar) is operated with the right stick. 
There are even moments when both cursors appear at the same time

- There is a lack of priority: In very  busy panels such as Character sheet, where there are many things to do, the behave of the new cursor is the expected, it starts at the top left and from there you take it where you want. 
But there are windows, like the "npc quest giver panel" where before opening the window itself the user knows what he wants to do, (accept or decline the quest, but most of the  times accept) here the cursor appears over the cancel button, and takes at least 3 moves and 1 click to accept the quest. That's a lot.
Before this patch you got controller shortcuts attached to the accept/decline buttons wich made the interaction with the panel lightning fast.

Here's a window by window review:

Game menu:
- I expected the mouse to switch columns if i kept pressing down at the bottom of the left column.

- With this new system, the feature of the on screen controller where you can hover each button to show you wich actionbars has assigned it almost disapears to the user knowledge as the new cursor doesnt move over the on screen controller buttons.

Suggestion: move the on screen controller under a tab in the "Controller" configuration panel. That way maybe it is possible to make the new cursor move through the controller buttons and higlight each actionbar buttons.

Warning: i dont think it is a good idea to be able to keep the controller on the menu screen and make it interactive for the new cursor. 
In that screen, and in order of importance i first expect a fluent navigation trough the menu buttons, and later a navigation through the controller buttons. So, pressing right should bring you to the right column, and not to any conroller button.

Controller configuration panel: 
The new cursor disappears, the old when joins appears.

Interface:
Here i almost had an orgasm, very fluent navigation. The sliders are not interactive for the new cursor tho.

Macros:
Lots of "false" possible buttons, the new cursor suggests positions where there are no buttons.

Addons: perfect

Character info:
Good navigation. 
I can pick objects and the old cursor appears to move the object wherever i want.
If i cancel or complete the action, im left with two cursors on screen.

Suggestion: could it be possible to auto hide again the old cursor and make it enter in "view mode" as soon as you complete or cancel the action of moving an object?

That way i t would feel like the second cursor only appears when needed, and disappears when done.

Spellbook:
Good navigation and same isseu as character info.

Achievements: 
Some false possible buttons

Bags
Curious behave, while moving the cursor through objects.
It gives me a live view of the object specification (expected)
But when browsing through categories i have to click on each category to see what objects are inside. I expected it to have a live view also.

When picking an object, i can move it but not drop it (erase it)

In the grid view, when highlighting each bag slots, the very first slot gets higlighted

Quest givers:
Here is the only thing i felt i didnt like it at all.
Before this updat there where those autobinded buttons that made the process very quick, accept or decline quests, also if im not wrong i was able to scroll using the arrows.

Now to accept a quest, it requires at least 3 mouse moves and one click to do a very common task when before only required one click.

Suggestion:
Maybe you could do an hybrid of the two concepts, where the square and triangle buttons serve as super quick shortcuts for important actions such as "accept"-square button or "cancel/close window"-triangle button.

That way you will have a key that will close your current window or cancel the current interaction (not accepting quest), and another button to accept quests or actions.

**Comment by OxydBCN** [comment]
i like to make a request. 
I'd like to see a lite version of the addon featuring only the movement/action bar interaction, basic bindings on important buttons and keep all the wow frames as default.

Why?  over the time, i had to finally play with a keyboard near me because ppl thought i were too antisocial.

So it is not much of a problem to me to navigate through default panels with the mouse / trackpad, and use the controller for basic important interactions.

I personally (and i know this is just me) i don't connect too much with the new bag system for example.

**Comment by seblindfors** [comment]
Great feedback! I will start to work on implementing a few changes to how it all works. I already have a system in place to ignore certain nodes if I deem them uninteresting, such as the close button on quest frames, since you already have the decline button there. It's not too tricky to make certain elements prioritized over others. I also gave the cursor a memory, which means it will return to the same node if the same frame is reopened without repositioning the cursor. 

Regarding the bag addon: it's buggy, incomplete and not really to my liking at all. I exclusively use the grid view atm, because it's more familiar. I'm going to move it out of the addon to a separate module, sort of like how DBM handles all its content modules. Creating a modular system also allows for other users to create plugins for the addon, to their own liking.

Using a modular approach, one can replace bits of the interface as they please or create more intuitively designed frames for things such as questing. This would also allow the user to disable the entire interface control, effectively creating a lite version of the addon.

**Comment by seblindfors** [comment]
Regarding the social bit, I was thinking of creating an optional virtual keyboard that pops up on the screen whenever an EditBox assumes keyboard focus. It might not even be too hard, using tables with locale translations for a ton of languages. It's on my bucket list, at least.

Regarding all the stuff you mentioned; I've been working on the UI navigation all day and made some changes to how it all works. The patch I just uploaded should be a lot more fluent and you won't have to press a button 4 times just to accept a quest anymore. 

I also decided to temporarily remove the bag module. It needs a lot more polishing before releasing, and when I do release it, it will be as a standalone addon that you can plug in to the new UI management system. I'm going to add a system to include whichever addons you want in the UI navigation shortly, meaning you can still modify your interface as you please but retain the functionality of the controller with those addons.

Here's a list of changes with the last update:
* Added support for additional frames (e.g. AuctionFrame, ChatMenu, Barbershop)
* Added support for scroll frame sliders 
* Added CheckBox object type to accepted nodes
* Added system for prioritizing nodes
* Improved overall UI navigation
* Increased dummy cursor speed when arrow button is held down
* Mouse look now stops when "Interact with mouseover" is triggered
* Cursor will now readjust itself or find another node if current node is obscured or lost
* Bag module has been scrapped

**Comment by OxydBCN** [comment]
You know, recently i decided to focus my work towards usability and interface. And if you tell me that you had a motivated day of work cause of this review, nothing makes me happier.

Gonna test the new patch right now!

PS: can you show me the model of your BT dongle? i'm tired of living in the 90's playing with a wired controller

**Comment by OxydBCN** [comment]
Btw, the patch is not up?

**Comment by seblindfors** [comment]
Weird, seems the github app didn't sync properly. Download here instead:
http://www.wowinterface.com/downloads/dl23536/ConsolePort-0.5.2.zip?144073362815

**Comment by seblindfors** [comment]
The patch is up now. Weirdly enough, my github app said it was pushed 12 hours ago.

Anyway, I'd like to mention one thing that isn't super intuitive. If you focus a scroll bar, you'll notice that the cursor indicator turns to L1. This is supposed to be the shift modifier. When held down, you can scroll using the up and down arrows. This is a compromise between always letting up/down scroll on frames and giving the user full control of the frame, including the up/down buttons.

**Comment by OxydBCN** [comment]
Hello old bags!!

Regarding the L1 new feature and the chance of getting two cursors, i like to suggest something to see if it likes you. Is this a thing?

![01](https://cloud.githubusercontent.com/assets/11269504/9560879/b952ede4-4e2d-11e5-909d-84cc8b8305c2.jpg)

![02](https://cloud.githubusercontent.com/assets/11269504/9560881/c5202344-4e2d-11e5-9b5a-efaa9db24e9f.jpg)

![03](https://cloud.githubusercontent.com/assets/11269504/9560883/d32647a2-4e2d-11e5-9dea-64d00cd9bbc4.jpg)

What do you think?

Btw, there is a kind of weird icon that appears when i press L1 (on scroll bars) or "O" in any other place.

Now im going to play a bit a check how fluent everything is.

**Comment by seblindfors** [comment]
Yeah, the weird icon was my rubbish scroll button texture. It looks horrible and I'm not afraid to say it. Made it in like 20 seconds because I needed a texture for it. 

I kinda like what you have in mind there, but there is a problem. The yellow tint backdrop is not good. Some buttons are small and obscure enough as is, so no need to add extra stuff onto them.

The border is good, but there's another problem with that. Not all buttons are the same size as their visual boundaries. There is no rule that says the various textures on a button have to intersect with the edge of the frame perfectly. Sometimes the button may actually be bigger than what is visible, and sometimes it may be smaller. The current approach of the cursor is to put it in the very center of the focused object. Drawing a border around objects may look very strange and out of place in a handful of  spots across the interface.

**Comment by OxydBCN** [comment]
I see.

**Comment by seblindfors** [comment]
Using /framestack, you can clearly see the issue:
![image](https://cloud.githubusercontent.com/assets/7464938/9561071/a1ce5e2a-4e36-11e5-8cfb-4cd9f8ebf25a.png)
![image](https://cloud.githubusercontent.com/assets/7464938/9561073/b4e2561a-4e36-11e5-931f-23c1dd2b0f34.png)
![image](https://cloud.githubusercontent.com/assets/7464938/9561079/d31a0e7a-4e36-11e5-8dba-5f841f6605dc.png)
![image](https://cloud.githubusercontent.com/assets/7464938/9561083/09cf303a-4e37-11e5-9122-4e1610c6b85d.png)

**Comment by seblindfors** [comment]
I took your advice and moved the controller out of the game menu. I will also be replacing the faux buttons on the game menu with textured buttons instead of text.

Here's what I'm working on now though:
![image](https://cloud.githubusercontent.com/assets/7464938/9561110/6050da16-4e38-11e5-836d-db7d5b1e9f67.png)

Do you have any thoughts on how it should work?

**Comment by OxydBCN** [comment]
I see the issue :(

On the second screen, and by the heriarchy order, it suggests that now the grid of buttons and combos is left, and you'll use the controller image instead, and be able to reconfigure the bindings from there?

**Comment by seblindfors** [comment]
Yes. Basically, you should be able to click on any of the buttons (doesn't matter if using fake cursor or real cursor) and then rebind from a flyout menu or something similar. I think this will add more clarity to the process of rebinding and managing the addon.

**Comment by OxydBCN** [comment]
I agree, and also if you keep the flashing action bar buttons when hovering each controler button, will add some visual guidance on where actually your bindings fall and to decide if it pleases your likings.

**Comment by OxydBCN** [comment]
Maybe i find some lack of help, like a small text telling a it the use of that screen, like "Hover over controller buttons to reveal its bindings, or rebind them pressing O"

**Comment by seblindfors** [comment]
Regarding the flashing buttons: I removed them for a very technical reason. They caused taint issues, because protected code runs through those functions every now and then. This meant that hovering the button on the game menu could create strange problems, like not being able to add a glyph or dismiss your pet. I'm going to copy the animations and textures soon and add it back in, but for now I've left them out because those functions are dangerous.

Regarding the binding frame: Yes, such text strings could be added above the controller.

**Comment by kilabyte** [comment]
i got everything up and working as per the readme however none of my square triangle etc action buttons work. So i can see in the action bar the item highlighting when i press the button but the action is never executed.

**Comment by seblindfors** [comment]
Here are some dumps of the new binding thingy:
![image](https://cloud.githubusercontent.com/assets/7464938/9570022/8f191aa2-4f7d-11e5-9f40-05ffad107919.png)
![image](https://cloud.githubusercontent.com/assets/7464938/9570024/96fbe8b2-4f7d-11e5-81ea-03256e82db85.png)
![image](https://cloud.githubusercontent.com/assets/7464938/9570026/9da358a8-4f7d-11e5-86f9-c89266e4d133.png)
![image](https://cloud.githubusercontent.com/assets/7464938/9570029/a3aac8f8-4f7d-11e5-8fbf-97250b26b029.png)

**Comment by OxydBCN** [comment]
Dat smells good!
Is up now?

**Comment by seblindfors** [comment]
On GitHub, yes.

**Comment by seblindfors** [comment]
Alpha 0.6.2
- Increased opacity on faded frames to 75% (up from 50%)
- New binding manager added to interface options
- New modified game menu
- Removed controller texture from game menu
- Removed popup system in favor of UIControl
- Simplified frame checking system
- Resolved taint issue when rebinding action buttons
- Fixed a couple of unintentional globals
- Fixed bug where secure action buttons would get incorrect actionpage attribute on login
- Fixed bug preventing item destruction popup from appearing

**Comment by kilabyte** [comment]
Update looks awesome! Was wondering if in a future release if you could have the option for a mouse reset location? For example i would always like the mouse to reset to the top of my characters head every time the mouses goes out and in focus (right click).

**Comment by OxydBCN** [comment]
:3

**Comment by seblindfors** [comment]
@kilabyte Can't do that within WoW, unfortunately.

**Comment by OxydBCN** [comment]
@seblindfors 
Did the last patch screw up the addon?

Mine is not working, and i inly user consoleport and dominos

**Comment by seblindfors** [comment]
@OxydBCN Not to my knowledge. Did you get any lua errors?

**Comment by OxydBCN** [comment]
Gonna check it again later, but the other day, right after the patch, the new virtual cursor was gone, and no bindingd at any actionbar button, plus the buttons to open game menu or map didnt work either. 

All the weird stuff is always for me :)

**Comment by OxydBCN** [comment]
But this is not gona chance the fact that thsnks to you i could still play while homdinh my baby :D

**Comment by seblindfors** [comment]
I uploaded a broken patch by mistake, but it was corrected shortly after. The one that is up now should work, but in case it doesn't I'm going to put up another one in a few minutes.

**Comment by OxydBCN** [comment]
Maybe that was the thing. But in the first moment after downloading i could test the new controller remaping and all went fine. Gonna try to download  this one

**Comment by seblindfors** [comment]
New patch is up, should work just fine.

**Comment by seblindfors** [comment]
@OxydBCN I was wondering if you might be interested in creating some hi-res button graphics for the addon? Having moved away from the old system of placing "guides" everywhere, using the flat graphics for large textures is no longer needed.

**Comment by seblindfors** [comment]
Latest update adds addon support to the virtual cursor:
![image](https://cloud.githubusercontent.com/assets/7464938/10085007/ad408c80-6306-11e5-8135-bf5fcfa32b42.png)

**Comment by OxydBCN** [comment]
Sounds good! What's your idea?
Any hint on the style? Size? 
Having two sizes,means i can do a set of small icons for the action bars a little more pixel art-ish, so they are more crisp and defined?

**Comment by seblindfors** [comment]
@OxydBCN Right now the size is fine. They are 64x64 which could arguably provide a lot of detail. The small buttons, however, only need to be 16x16, since their current size is 14x14 in game. (remember they have to be a power of 2). 

And yes; the small icons can be small and crispy, while the large 64x64 buttons can have a lot of detail and nice finish. Feel free to be creative with designs and color choices.

**Comment by kilabyte** [comment]
why the removal to work with Storyline addon? The macro you gave me as well doesn't work now under this new patch?

**Comment by seblindfors** [comment]
Face lift!
![image](https://cloud.githubusercontent.com/assets/7464938/10549597/58ccbde8-7442-11e5-90f0-7da0eee8d58d.png)

**Comment by ddormer** [comment]
Support for ElvUI's party frames would be amazing. At the moment "Toggle raid cursor" doesn't work if ElvUI's unit frames are enabled.

**Comment by seblindfors** [comment]
Hm. It works with default frames, Grid and Zpearl. Curious as to why it doesn't work with ElvUI. WIll look into it.

**Comment by OxydBCN** [comment]
Didn't had the time to test the new improvements until today, and oh my god, how it has grown!! Loving the new typing method. Love the new addon interface. 

The only ting i like only a little less is that thing about mking the cursor bigger each time it changes a link.

Seb, i'm amazed of what you are doing!

**Comment by seblindfors** [comment]
Thanks man :+1: 
Sorry I phased out some of the graphics you put work into, but it was all in the name of progress.
I know you've been busy as of late, but if you feel like putting your two cents in, your contributions are always welcome. :stuck_out_tongue_winking_eye:

**Comment by seblindfors** [comment]
@ddormer ElvUI is doing something shady that messes up the frame caching in the secure environment. This is not a problem with ConsolePort, but rather with ElvUI. I suggest disabling the ElvUI raid frames and using Grid instead. This is out of my hands.

**Comment by OxydBCN** [comment]
Here i come with another little review.

I have to say that this addon has grown very well!

Then...

Learning curve:
Ok, now this addon is big, and it has many many features, the utility belt, the keyboard.
I missed some lack of guidance in the use of some features, or at least how to pop them.

Overall, for my likings, i feel there is way too much attention seekers.
The new behave of the action bar icons, the virtual cursor summoning animation... There is a lot of movement everywhere. Not excesively distracting but on my opinion it feels unnecesary when you want to pay attention to the battlefield.

I'd suggest a "boring mode" where you can disable all the animations, for when you are used to the addon.

The new behave of the action bars:
I created a character to test how it feels on a new player with an empty toolbar. 
This addon by default has a pretty beautiful keybinds. Your first spells lay on the AOXO buttons. This is what any hooman would expect. Some spell lays on the R1 making you a hint that those keys works aswell.
As the character grows you will see that there are the modifiers and the arrow keys, ams then you are prepared to understand the full potential of the addon.

I see you did this nnew behave to help players understand what keys where available when pressing a modifier. Is a good idea, but in an empty toolbar it feels broken.

If i press a modifier on a level 1 warrior, the keybindings disappear.
Maybe instead of making dissapear all the icons except for the ones that are linked to the modifier you are pressing, you could play with opacity, lowering drastically those icons unavailable.

This way we would have less surprising movement.

Side note: i have to say that i can be more happy with you maintaining the default bag system.

Regarding bags.
- I love the way i can link to utility belt
- i had trouble figuring out how to equip an item:
The cursor says i can press circle, but if i do, it goes to the utility belt.
Then i had to realize that is pressing square what allows you to equip

Then at this point i suggest the use of the mini icons to place a little list of the options of each item on the cursor itself instead of the circle button. With a dark background, like the new buttons, and the text color matching the button color.

That way you can cover all the actions you can do to an item from the bag itself.


Ill continue later....

**Comment by OxydBCN** [comment]
Btw, what part of the addon you are working on actually? I ask to review over that part specifically.

Btw, the keyboard is amazing, there is absolutely nothing bad  i could say about it.

I inly miss a hint to know how to summon it. (Btw, there is no way to summon the keyboard without actually pressing the keyboard? That is sooo 90's...)

I miss a hint to know how to use the autofill function. 

Other than that, its amazing. Super fast learning curve

**Comment by seblindfors** [comment]
Opacity on animated aciton bars only adds to the confusion. I tried that out because I was aware of the problem on low level characters, but it didn't help. Moderate opacity is confusing and extremely low opacity looks like out of place because they're placed on the upper right corner and not dead center. Animation will be optional in the next patch or the patch after that. I'm also going to add more hotkey settings along with more icon sets to choose from. The old set with the new set's drop shadow looked kinda cool actually, so I'll be adding that too.

I've made some changes to the cursor for the upcoming patch. One of the spell effects was removed because it was a little too much, and the remaining spell effect will calculate its animation cycle based on the distance between the two nodes it's jumping between. This also applies to the scaling animation. This should reduce the amount of "flashing" when moving the cursor, especially between nodes that are positioned directly next to each other. 

I'm planning on adding a tutorial of sorts, or maybe just a popup that has something like "Tip: You can double click the utility belt binding to toggle its visibility". That would allow me to introduce some of the more complex controls without bloating the screen with permanent instructions. 

I can't check whether a button can be right-clicked, which is why there's no indication of that in the bags. There used to be a tooltip addition that said "Equip", but it was removed because of changes to tooltip management. I'll add that to my list of things to do. 

Keyboard instructions can be seen in the config and the keyboard itself can be opened by binding "Open Chat" (default binding is L1+L2+PS).

**Comment by seblindfors** [comment]
Old style:
![old_ani](https://cloud.githubusercontent.com/assets/7464938/12036861/20bdc40a-ae4a-11e5-8ce6-716a9d684395.gif)

New style:
![new_ani](https://cloud.githubusercontent.com/assets/7464938/12036864/29480d56-ae4a-11e5-94a4-cbae404087eb.gif)

**Comment by OxydBCN** [comment]
I like it

**Comment by OxydBCN** [comment]
I have a request! (dunno if "suggestion" is a better word for this)

Well, the case is that when i'm flying with a mount, i usually like to look around to avoid dying in boredom.
With console port this is not possible because the right stick makes you turn your mount.

would it be possible to use the right stick with "L1" while flying to make the camera move around you?
Then if you release "L1" the right stick would get its normal behavior controlling the direction of the mount.

What do you think?

**Comment by OxydBCN** [comment]
A suggestoin regarding the recent cast circle on the raid cursor.

What do you think about using an horizontal cast bar than a circle one?
I mean adding a very tiny and minimalistic bar underneath the raid cursor circle.
A casting bar vidually like the one that uses Apple:
This:
![unovp](https://cloud.githubusercontent.com/assets/11269504/12372021/4293fa2a-bc48-11e5-814e-95f5192e6784.png)
or this:
![apple_logo](https://cloud.githubusercontent.com/assets/11269504/12372023/4858a65e-bc48-11e5-9694-0424275ae3dc.png)

I suggest this to make it more readable.

Also actually the cast circle sometimes finishes a loop, sometime it goes further than a loop.

**Comment by seblindfors** [comment]
Maybe. I kinda like the circle though :wink: 
When the cast goes further than a loop, it means you have some minor lag. I didn't fix it because I thought it was kind of a cool unintended feature. 

I kinda want the cast bar to be visually noticeable though so you can still see it clearly in the corner of your eye if you're not staring at your unit frames. If you can give me a mockup with this type of cast bar that fits the criteria then I might change it. :+1:

**Comment by OxydBCN** [comment]
Announcing lag happening is a curious feature that you could explode more.

The little moment i tried the new castbar (without a deep testing) i felt it was not a bit unreliable. I didn't connect much with the lag feature, wich is interesting, but for a person who don't understand shit, i found it a bitn unreliable, as some casts where finished before a loop and some after the loop.

At this point i'd like to ask:
What is the important thing you need to see when casting but not staring at the Raid Cursor? 
The moment when the spell is succesfully casted?

Maybe you could add (to the current cast) a slight flash that tells you the exact moment you can cast again,
And for the lag thin, maybe you could turn the cast bar into red once it goe far from 1 loop and turn the flash red.

I could do a mockup of the alternative castbar, but only once you consider that part needs a refresh. 
What you have now is not broken.

**Comment by seblindfors** [comment]
The import thing you need to see is that you are casting and the color of the cast bar, since you want to know which party member you're casting on. When you're looking directly at it, you will only get a general idea when to queue the next spell, which has worked out well enough for me personally as a healer. 

I actually did try to stop the looping at first, but it looked weird, like it was suddenly broken. The cast bar currently looks fluid and smoothly animated, but completely stopping the loop once it has finished a cycle didn't fit it. The default Blizzard cast bar just stops at the end and holds there until the cast is finished, which is when it changes color, but that is visually sound for a horizontal status bar. Not so much for a looping circle. 

If you don't want to create a mockup out of the blue, I'd say adding a flash or indicator of successful cast would be a better way to go, since I kind of like the current animation.

**Comment by OxydBCN** [comment]
Dont get me wrong, i would do a three hundred different mockups if you like
it. I said i wuoldnt do it to avoid giving you the wrong idea that the
current want doesnt works. Cause it does work

**Comment by seblindfors** [comment]
Regarding our bnet discussion earlier, I put together a video of doing the bronze proving grounds as a level 90 disc priest, the same setup that you had issues with. As you can see in the video, it works just fine on my end. Still not sure how to reproduce the problem you had.
https://youtu.be/TRVcesPkYV8

**Comment by OxydBCN** [comment]
Good to know. Gonna try more stuff and see.

**Comment by OxydBCN** [comment]
AWW YIISS!

Just tested and now it runs better!
Wanted to point a couple of things:

In your video, you can apply any healing spell to anyone.
In my case I cannot apply Shield or Flash heal. Each time i cast them they land on me, no matter what.
 Disabling Bartender solves the problem.

On the other hand, i find that the position of the raid cursor, or its size are a bit intrusive, leaving me without much room to see the health of my poor fellows.
see:
![wowscrnshot_011816_083228](https://cloud.githubusercontent.com/assets/11269504/12385875/aca50f0c-bdc0-11e5-9c50-829c1cb2e3f5.jpg)

Sidenote: God bless the Nelf Redesing :D

**Comment by OxydBCN** [comment]
Found out something more regarding this topic.
Whenever i have a spell that works with the raid cursor placed on another bar that is not the ActionBar 1. It just doesn't work.
Note that in my last screenshot, Flash Heal and Shield are placed into ActionBar 6.

I tried all the other bars and the same result, except for ActionBar 1

Here's how i sorted my bars  (weirdly sorted)
![wowscrnshot_012116_022750](https://cloud.githubusercontent.com/assets/11269504/12468854/7d233136-bfe7-11e5-848e-f7f8faf9c96e.jpg)

**Comment by seblindfors** [comment]
Getting real tired of this Bartender bullshit.

**Comment by OxydBCN** [comment]
:D
Another option is to uninstall Bartender.
I think i opt for that, for the sake of how i apreciate you

**Comment by seblindfors** [comment]
Nah, it will be fixed. Somehow. It just feels like you fix something and then break something else. Pretty much have to fix it, since a lot of people prefer Bartender over default UI.

**Comment by OxydBCN** [comment]
I know this feeling bro.
![unnamed](https://cloud.githubusercontent.com/assets/11269504/12479456/68b96932-c03b-11e5-988d-58ecd6e08d19.gif)
